### PR TITLE
Implement SplashAttacks Meta tags visualization and Added new report

### PR DESCRIPTION
**Pull request by Klissans**
* implemented cone/corridor visualization for SplashAttacks meta tags
* added a new report which dumps all .meta and anim headers as json files in organized folders

### CODE REVIEW

File: View3D/Rendering/RenderItems/ConeRenderItem.cs [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+        public void Draw(GraphicsDevice device, CommonShaderParameters parameters)
+        {
+            _shader.Parameters["View"].SetValue(parameters.View);
+            _shader.Parameters["Projection"].SetValue(parameters.Projection);
+            _shader.Parameters["World"].SetValue(ModelMatrix);
+        
+            foreach (var pass in _shader.CurrentTechnique.Passes)
+            {
+                pass.Apply();
```
Comment (olekristianhomelien) [inline View3D/Rendering/RenderItems/ConeRenderItem.cs:113]: I think you can change this to just upload a list of points to the card and save the loop. 
Have a look at LineMeshRender.cs

if applicable, could this be solved by adding a new function like the void AddCube(Matrix transform, Color color) function?

File: View3D/Rendering/RenderItems/CorridorRenderItem.cs [hunk 2]
```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using Microsoft.Xna.Framework.Graphics;
+using System.Linq;
+using SharpDX;
```
Comment (olekristianhomelien) [inline View3D/Rendering/RenderItems/CorridorRenderItem.cs:4]: is sharddx needed?

File: View3D/Rendering/RenderItems/CorridorRenderItem.cs [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        }
+
+        public CorridorRenderItem(Effect shader, Vector3 startPos, Vector3 endPos, float width, Color color) : this(shader, startPos, endPos, width)
+        {
+            _colour = color;
+        }
+        
+        private void Init()
+        {
+            _halfWidth = _width / 2;
```
Comment (olekristianhomelien) [inline View3D/Rendering/RenderItems/CorridorRenderItem.cs:43]: if applicable, could this be solved by adding a new function like the void AddCube(Matrix transform, Color color) function?

File: View3D/Rendering/RenderItems/LineSegmentRenderItem.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
+﻿using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using System.Linq;
+using View3D.Components.Rendering;
+
+namespace View3D.Rendering.RenderItems
+{
+    public class LineSegmentRenderItem: IRenderItem
```
Comment (olekristianhomelien) [inline View3D/Rendering/RenderItems/LineSegmentRenderItem.cs:8]: There is already a linemeshrenderer

File: View3D/Rendering/RenderItems/ConeRenderItem.cs (hunk 1 above)
Comment (Klissans) [reply View3D/Rendering/RenderItems/ConeRenderItem.cs:113]: Whoops, if-statement should be outside of the inner for-loop, just noticed.

> I think you can change this to just upload a list of points to the card and save the loop.
I can replace `_circles `with 1d array with the size twice as large and pass it to `device.DrawUserPrimitives` with `PrimitiveType.LineList `argument. Will it be faster?

Comment (olekristianhomelien) [reply View3D/Rendering/RenderItems/ConeRenderItem.cs:113]: device.DrawXXX is super slow. Not a huge issue for a simple case like this, but I think its among the slowest calls in a game engine and one typically goes to (extreme) lengths to minimize the call count.

**Review by olekristianhomelien — APPROVED** [review]

**Pull request merged**